lib/helpers/topic.js

```diff
@@ -203,48 +211,53 @@ function parseTopicFromContentfulEntry(contentfulEntry) {
 
   const data = helpers.contentfulEntry.getSummaryFromContentfulEntry(contentfulEntry);
   data.postType = module.exports.getPostTypeFromContentType(contentType);
+
+  // Get campaign if topic has a campaign set.
   const campaignConfigEntry = contentfulEntry.fields.campaign;
-  let promise = Promise.resolve(null);
-  let campaignId;
-  if (campaignConfigEntry) {
-    campaignId = contentful.getCampaignIdFromContentfulEntry(campaignConfigEntry);
-    promise = helpers.campaign.fetchById(campaignId);
-  }
-  return promise
-    .then((campaign) => {
-      data.campaign = campaign;
-      data.templates = module.exports
-        .parseTopicTemplatesFromContentfulEntryAndCampaign(contentfulEntry, campaign);
-      return data;
-    })
-    .catch((error) => {
-      if (error.status === 404) {
-        // The campaignId is set via freeform text field on the campaign content type.
-        // If a topic's campaign is misconfigured, we still want to return topic data.
-        // Hardcode the status as closed to auto reply with the campaignClosed template.
-        data.campaign = { id: campaignId, status: 'closed' };
-        data.templates = module.exports
-          .parseTopicTemplatesFromContentfulEntryAndCampaign(contentfulEntry, null);
-        return data;
-      }
+  try {
+    data.campaign = campaignConfigEntry ? await helpers.campaign
+      .getById(contentful.getCampaignIdFromContentfulEntry(campaignConfigEntry)) : {};
+  } catch (error) {
+    // The campaignId is set via freeform text field on the campaign content type.
```

Comment:
This line can be improved by being more explicit about where this things are happening, example: `The campaignId value is set via manual input on the campaign content type entry using the Contentful UI`